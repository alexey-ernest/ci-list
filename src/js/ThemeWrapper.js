/**
 * Wraps component with theme provider to avoid Material UI errors.
 */

import React, {Component, PropTypes} from 'react';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {
  lightBlue500,
  lightBlue700,
  grey400,
  pinkA200,
  grey200,
  grey500,
  darkBlack,
  white,
  grey300,
  fullBlack
} from 'material-ui/styles/colors';
import {fade} from 'material-ui/utils/colorManipulator';


// More on Colors: http://www.material-ui.com/#/customization/colors
var muiTheme = getMuiTheme({
  palette: {
    primary1Color: lightBlue500,
    primary2Color: lightBlue700,
    primary3Color: grey400,
    accent1Color: pinkA200,
    accent2Color: grey300,
    accent3Color: grey500,
    textColor: darkBlack,
    alternateTextColor: white,
    canvasColor: white,
    borderColor: grey300,
    disabledColor: fade(darkBlack, 0.3),
    pickerHeaderColor: lightBlue500,
    clockCircleColor: fade(darkBlack, 0.07),
    shadowColor: fullBlack,
  }
});

export default class ThemeWrapper extends Component {

  static childContextTypes = {
    muiTheme: React.PropTypes.object.isRequired
  };

  getChildContext() {
    return {muiTheme: muiTheme};
  }

  render() {
    return (
      <MuiThemeProvider muiTheme={muiTheme}>
        {this.props.children}
      </MuiThemeProvider>
    );
  }
}
