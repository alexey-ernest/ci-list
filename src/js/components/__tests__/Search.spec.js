
import React from 'react';
import renderer from 'react-test-renderer';

import ThemeWrapper from '../../ThemeWrapper';
import Search from '../Search.react';


jest.mock('react-dom');

describe('Search component', () => {

  it('should render correctly', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <Search />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should update value by onChange event', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <Search />
      </ThemeWrapper>
    );

    let tree = component.toJSON();

    // manually trigger the callback
    tree.children[0].children[0].children[0].children[1].props.onChange({
      target: {
        value: 'a'
      }
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onSubmit handler if <Enter> key pressed', () => {
    let onSubmit = jasmine.createSpy();

    const component = renderer.create(
      <ThemeWrapper>
        <Search onSubmit={onSubmit} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();

    // type some text
    tree.children[0].children[0].children[0].children[1].props.onChange({
      target: {
        value: 'alex'
      }
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // type space
    tree.children[0].children[0].children[0].children[1].props.onKeyDown({
      keyCode: 32
    });

    // type enter
    tree.children[0].children[0].children[0].children[1].props.onKeyDown({
      keyCode: 13
    });

    expect(onSubmit.calls.count()).toBe(1);
    expect(onSubmit.calls.argsFor(0)[0]).toBe('alex');
  });
});
