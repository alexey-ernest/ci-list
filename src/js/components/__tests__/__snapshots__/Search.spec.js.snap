exports[`Search component should call onSubmit handler if <Enter> key pressed 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 8
      }
    }>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16
        }
      }>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%"
          }
        }>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms"
            }
          }>
          Filter builds
        </div>
        <input
          disabled={false}
          id="keywords"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onFocus={[Function anonymous]}
          onKeyDown={[Function anonymous]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "initial",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%"
            }
          }
          type="text"
          value="alex" />
        <div>
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 1px",
                "borderColor": "#e0e0e0",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%"
              }
            } />
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 2px",
                "borderColor": "#03a9f4",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%"
              }
            } />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search component should render correctly 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 8
      }
    }>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16
        }
      }>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%"
          }
        }>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms"
            }
          }>
          Filter builds
        </div>
        <input
          disabled={false}
          id="keywords"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onFocus={[Function anonymous]}
          onKeyDown={[Function anonymous]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "initial",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%"
            }
          }
          type="text"
          value="" />
        <div>
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 1px",
                "borderColor": "#e0e0e0",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%"
              }
            } />
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 2px",
                "borderColor": "#03a9f4",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%"
              }
            } />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search component should update value by onChange event 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 8
      }
    }>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16
        }
      }>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 48,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%"
          }
        }>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms"
            }
          }>
          Filter builds
        </div>
        <input
          disabled={false}
          id="keywords"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onFocus={[Function anonymous]}
          onKeyDown={[Function anonymous]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "initial",
              "font": "inherit",
              "height": "100%",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%"
            }
          }
          type="text"
          value="a" />
        <div>
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 1px",
                "borderColor": "#e0e0e0",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%"
              }
            } />
          <hr
            style={
              Object {
                "border": "none",
                "borderBottom": "solid 2px",
                "borderColor": "#03a9f4",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%"
              }
            } />
        </div>
      </div>
    </div>
  </div>
</div>
`;
